### PR TITLE
Improve e2e test reliability

### PR DESCRIPTION
**Pull request by TheRealJon**
Improves reliability of e2e tests. 

- Increase specificity on a few ambiguous element locators. 
- Add a condition to check for `.co-m-loader` elements on a page so that we can wait until they are all gone before proceeding.
- Fix a few possible async race conditions
- Use better names for created resources so that they can be more easily selected

### CODE REVIEW

File: frontend/integration-tests/tests/crud.scenario.ts [hunk 1]
```diff
@@ -87,28 +87,30 @@ describe('Kubernetes resource CRUD operations', () => {
...
+
+        const yamlLinkIsPresent = await crudView.createYAMLLink.isPresent();
+        if (yamlLinkIsPresent) {
+          await crudView.createYAMLLink.click();
         }
         await yamlView.isLoaded();
 
         const content = await yamlView.editorContent.getText();
-        const newContent = _.defaultsDeep({}, {metadata: {name: testName, labels: {[testLabel]: testName}}}, safeLoad(content));
+        const newContent = _.defaultsDeep({}, {metadata: {name: `${resourceName}`, labels: {[testLabel]: testName}}}, safeLoad(content));
```
Comment (TheRealJon) [inline frontend/integration-tests/tests/crud.scenario.ts]: oops, don't need string interpolation on  `resourceName` here

File: frontend/integration-tests/tests/crud.scenario.ts [hunk 2]
```diff
@@ -164,17 +164,40 @@ describe('Kubernetes resource CRUD operations', () => {
...
     it('creates a RoleBinding', async() => {
+      await browser.wait(crudView.untilNoLoadersPresent);
+
+      // Input binding name
       await $('#role-binding-name').sendKeys(bindingName);
-      await $('#ns-dropdown').click().then(() => browser.actions().sendKeys(testName, Key.ARROW_DOWN, Key.ENTER).perform());
-      await $('#role-dropdown').click().then(() => browser.actions().sendKeys('cluster-admin', Key.ARROW_DOWN, Key.ENTER).perform());
+
+      // Select namespace
+      await $('#ns-dropdown').click();
```
Comment (alecmerdler) [inline frontend/integration-tests/tests/crud.scenario.ts]: Can we pull this out to a `selectFromDropdown` function?

File: frontend/integration-tests/views/yaml.view.ts [hunk 3]
```diff
@@ -1,16 +1,21 @@
 /* eslint-disable no-undef, no-unused-vars */
-
-import { $, $$, by, Key, browser, ExpectedConditions as until } from 'protractor';
+import { $, by, Key, browser, ExpectedConditions as until } from 'protractor';
+import { untilNoLoadersPresent } from './crud.view';
```
Comment (alecmerdler) [inline frontend/integration-tests/views/yaml.view.ts]: Not sure how I feel about different views importing from each other.

File: frontend/integration-tests/tests/crud.scenario.ts [hunk 4]
```diff
@@ -295,20 +314,19 @@ describe('Kubernetes resource CRUD operations', () => {
     beforeAll(async() => {
       await browser.get(`${appHost}/k8s/ns/${testName}/${plural}/new`);
       await yamlView.isLoaded();
-
       const content = await yamlView.editorContent.getText();
-      const newContent = _.defaultsDeep({}, {metadata: {name, namespace: testName}}, safeLoad(content));
-      await yamlView.setContent(safeDump(newContent));
+      const newContent = _.defaultsDeep({}, {metadata: {name, namespace: testName}}, await safeLoad(content));
+      await yamlView.setContent(await safeDump(newContent));
```
Comment (spadgett) [inline frontend/integration-tests/tests/crud.scenario.ts]: Aren't `safeLoad` and `safeDump` synchronous? I wouldn't expect to need `await` here

File: frontend/integration-tests/views/crud.view.ts [hunk 5]
```diff
@@ -16,7 +16,9 @@ export const cancelBtn = $('#cancel');
 /**
  * Returns a promise that resolves after the loading spinner is not present.
  */
-export const isLoaded = () => browser.wait(until.presenceOf($('.loading-box__loaded')), 10000).then(() => browser.sleep(1000));
+export const untilLoadingBoxLoaded = until.presenceOf($('.loading-box__loaded'));
+export const untilNoLoadersPresent = waitForNone($$('.co-m-loader'));
```
Comment (spadgett) [inline frontend/integration-tests/views/crud.view.ts]: 👍

File: frontend/integration-tests/views/yaml.view.ts (hunk 3 above)
Comment (TheRealJon) [reply frontend/integration-tests/views/yaml.view.ts]: Maybe we should move that export into the protractor conf file

**Comment by spadgett** [comment]
I believe this will fix https://jira.coreos.com/browse/CONSOLE-1232

**Comment by TheRealJon** [comment]
@spadgett @alecmerdler I believe I have addressed all of the feedback.

**Pull request merged**